src/devinterp/slt/llc.py

```diff
@@ -6,106 +6,120 @@
 from devinterp.optim.sgld import SGLD
 from devinterp.slt.callback import SamplerCallback
 from devinterp.slt.sampler import sample
+from devinterp.utils import (
+    optimal_temperature,
+    get_init_loss_one_batch,
+    get_init_loss_full_batch,
+)
 
 
 class LLCEstimator(SamplerCallback):
     """
-    Callback for estimating the Local Learning Coefficient (LLC) in a rolling fashion during a sampling process. 
+    Callback for estimating the Local Learning Coefficient (LLC) in a rolling fashion during a sampling process.
     It calculates the LLC based on the average loss across draws for each chain:
     $$
     TODO
     $$
-    
+
     Attributes:
         num_chains (int): Number of chains to run. (should be identical to param passed to sample())
         num_draws (int): Number of samples to draw. (should be identical to param passed to sample())
-        n (int): Number of samples used to calculate the LLC.
+        temperature (float): Temperature used to calculate the LLC.
         device (Union[torch.device, str]): Device to perform computations on, e.g., 'cpu' or 'cuda'.
     """
-    def __init__(self, num_chains: int, num_draws: int, n: int, device: Union[torch.device, str]="cpu"):
+
+    def __init__(
+        self,
+        num_chains: int,
+        num_draws: int,
+        temperature: float,
+        device: Union[torch.device, str] = "cpu",
+    ):
         self.num_chains = num_chains
         self.num_draws = num_draws
-        self.losses = torch.zeros((num_chains, num_draws), dtype=torch.float32).to(device)
+        self.losses = torch.zeros((num_chains, num_draws), dtype=torch.float32).to(
+            device
+        )
 
-        self.n = torch.tensor(n, dtype=torch.float32).to(device)
+        self.temperature = torch.tensor(temperature, dtype=torch.float32).to(device)
         self.llc_per_chain = torch.zeros(num_chains, dtype=torch.float32).to(device)
-        self.llc_mean = torch.tensor(0., dtype=torch.float32).to(device)
-        self.llc_std = torch.tensor(0., dtype=torch.float32).to(device)
+        self.llc_mean = torch.tensor(0.0, dtype=torch.float32).to(device)
+        self.llc_std = torch.tensor(0.0, dtype=torch.float32).to(device)
 
         self.device = device
 
-    def update(self, chain: int, draw: int, loss: float):
-        self.losses[chain, draw] = loss 
-
-    @property
-    def init_loss(self):
-        return self.losses[0, 0]
+    def update(self, chain: int, draw: int, loss: float, init_loss):
+        self.losses[chain, draw] = loss
+        self.init_loss = init_loss  # This is clunky, sorry
```

Comment:
what's the reason for changing this? (not saying the other way was graceful, just curious if e.g. this has any behavioral change or is useful for some other reason)